### PR TITLE
feat(SilentTyping): add toggle command and icon

### PR DESCRIPTION
**Pull request by dzshn**
adds a simple command `/silenttype` so you can toggle whether the plugin is enabled without having to restart

### CODE REVIEW

**Comment by Vendicated** [comment]
way cooler would be to add a chatbar button like invisiblechat

**Comment by dzshn** [comment]
I'm kinda against adding a button cuz it's not something you toggle often but yeah maybe could be an option

File: src/plugins/silentTyping.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+*/
+
+import { ApplicationCommandInputType, ApplicationCommandOptionType, findOption, sendBotMessage } from "@api/Commands";
+import { get, set } from "@api/DataStore";
+import ErrorBoundary from "@components/ErrorBoundary";
+import { Devs } from "@utils/constants";
+import { useForceUpdater } from "@utils/misc";
+import definePlugin, { OptionType } from "@utils/types";
+import { Button, ButtonLooks, ButtonWrapperClasses, FluxDispatcher, React, Tooltip } from "@webpack/common";
+import { Settings } from "Vencord";
```
Comment (Vendicated) [inline src/plugins/silentTyping.tsx:26]:
```suggestion
import { Settings } from "@api/settings";
```

**Comment by fawni** [comment]
if ur gonna add a button can u make its existence toggleable because i personally don't need to toggle it so a button would just be annoyance to me

**Review by Vap0r1ze — COMMENTED** [review]
hmm

**Comment by Vendicated** [comment]
are u two done now?

**Comment by dzshn** [comment]
I'd say so yes

**Comment by Vendicated** [comment]
oh yeah you should switch to vaps new settings api maybe

File: src/plugins/silentTyping.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    isEnabled: boolean;
+}
+const getSettings = () => Settings.plugins.SilentTyping as SilentTypingSettings;
+
+function SilentTypingToggle() {
+    const { isEnabled } = useSettings(["plugins.SilentTyping.isEnabled"]).plugins.SilentTyping as SilentTypingSettings;
+    const toggle = () => getSettings().isEnabled = !getSettings().isEnabled;
+
+    return (
+        <Tooltip text={isEnabled ? "Disable silent typing" : "Enable silent typing"} style={{ scale: "0.5" }}>
```
Comment (Vendicated) [inline src/plugins/silentTyping.tsx:38]: huh why this scale

File: src/plugins/silentTyping.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            {(tooltipProps: any) => (
+                <Button
```
Comment (Vendicated) [inline src/plugins/silentTyping.tsx:40]: you should wrap this in a div similar to https://github.com/Vendicated/Vencord/commit/19c9a132733dfb67de6d62d6da1348944cbe1146

File: src/plugins/silentTyping.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        {
+            find: ".activeCommandOption",
+            predicate: () => settings.store.showIcon,
+            replacement: {
+                match: /\i=\i\.activeCommand,\i=\i\.activeCommandOption,.{1,133}(.)=\[\];/,
+                replace: "$&;$1.push($self.chatBarIcon());",
+            }
+        },
+    ],
+    dependencies: ["CommandsAPI"],
```
Comment (Vap0r1ze) [inline src/plugins/silentTyping.tsx:90]:
```suggestion
    dependencies: ["CommandsAPI"],
    settings,
```

**Review by Vap0r1ze — CHANGES_REQUESTED** [review]

**Review by Vendicated — APPROVED** [review]

**Pull request merged**